timon/db/backends.py

```diff
@@ -87,25 +88,28 @@ def start(self, probenames):
     def stop(self):
         logger.info("Stopping Peewee backend")
         self.stopevent.set()
+        self.waitevent.set()
         self.thread_store.join()
         self.db.close()
 
+    def _flush(self):
+        """
+        Force the flush of the queue in db
+        """
+        self.waitevent.set()
+        self.waitevent.clear()
+
     def get_probe_results(self, probename):
         from timon.db import peewee_utils
         probe_results = []
+        self._flush()
+        cnt = 0
+        while self.thread_store.status != "WRITING" and cnt < 100:
+            # wait the lock is acquired by the thread
+            self.waitevent.wait(0.00001)
```

Comment:
this delay is probably too short. it's in the range of 10 microseconds. and will therefore not save a lot of CPU.
I think a millisecond is sufficiently small.

Reply:
please explain as well whether get_probe_results is being called by a thread or in an asyncio context.
as this is form an asyncio point of view a blocking function.

Reply:
Mmm, i think the microsecond is suficient to acquire a lock, i can change but it seems working

Reply:
let's discuss on teams.

such a small delay doesn't help the CPU a lot if the thread is really blocked.